### PR TITLE
In App Update Information Form

### PR DESCRIPTION
**Pull request by shubham1g5**
Spec: https://docs.google.com/document/d/1T2sdjD1QbAnTSdaqc6KLrS9jRqCf7M3hDC40pus4WzY/edit#

Adds ability to parse and launch Update Info Form on user login after update. 

cross-request: https://github.com/dimagi/commcare-core/pull/796

### CODE REVIEW

**Comment by shubham1g5** [comment]
@damagatchi retest this please

**Comment by shubham1g5** [comment]
@damagatchi retest this please

File: app/unit-tests/src/org/commcare/android/tests/processing/FormStorageTest.java [hunk 1]
```diff
@@ -307,10 +307,11 @@
             , "org.javarosa.xpath.expr.XPathDistinctValuesFunc"
             , "org.javarosa.xpath.expr.XPathSleepFunc"
 
-            // Added in 2.42
+            // Added in 2.44
             , "org.commcare.android.database.user.models.FormRecordV4"
```
Comment (amstone326) [inline app/unit-tests/src/org/commcare/android/tests/processing/FormStorageTest.java]: I don't think these other 3 classes (lines 311-313) were added in 2.44?

File: app/unit-tests/resources/commcare-apps/update_info_form/default/app_strings.txt [hunk 2]
```diff
@@ -0,0 +1,29 @@
+app.display.name=Update Info Form
```
Comment (amstone326) [inline app/unit-tests/resources/commcare-apps/update_info_form/default/app_strings.txt]: In the interest of keeping test resources as concise as possible, you should be able to delete both `app_strings` files.

File: app/unit-tests/resources/commcare-apps/update_info_form/media_suite.xml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+<?xml version='1.0' encoding='UTF-8'?>
```
Comment (amstone326) [inline app/unit-tests/resources/commcare-apps/update_info_form/media_suite.xml]: You can also delete this file as long as you remove the reference to it in the profile

File: app/src/org/commcare/activities/HomeScreenBaseActivity.java [hunk 4]
```diff
@@ -719,6 +739,7 @@ private boolean processReturnFromFormEntry(int resultCode, Intent intent) {
                 SqlStorage<FormRecord> formRecordStorage = CommCareApplication.instance().getUserStorage(FormRecord.class);
                 formRecordStorage.write(current);
                 checkAndStartUnsentFormsTask(false, false);
+                CommCareApplication.instance().getCommCarePlatform().setUpdateInfoFormXmlns(null);
```
[the remark this answers is not in the document]
Comment (shubham1g5) [reply app/src/org/commcare/activities/HomeScreenBaseActivity.java]: @amstone326 Makes sense. I was under the impression that user need to click finish and hence create and send the update info form instance as an evidence of him being through the form. 

One other question - Do we need to show the update info form just after the CommCare App update or also after CommCare App install ? I am thinking of scenario when some user uninstalled the ccz and updated directly to the latest version by installing the ccz again.

File: app/unit-tests/resources/commcare-apps/update_info_form/default/app_strings.txt (hunk 2 above)
Comment (amstone326) [reply app/unit-tests/resources/commcare-apps/update_info_form/default/app_strings.txt]: Ah yea you're right, I mis-remembered. Sorry about that.

**Pull request merged**